File: game/enemy.go
package game

import (
	"github.com/BigJk/end_of_eden/internal/lua/luhelp"
)

// Enemy represents a definition of a enemy that can be linked from a Actor.
type Enemy struct {
	ID          string
	Name        string
	Description string
	InitialHP   int
	MaxHP       int
	Gold        int
	Look        string
	Color       string
	Intend      luhelp.OwnedCallback
	Callbacks   map[string]luhelp.OwnedCallback
	Test        luhelp.OwnedCallback
	BaseGame    bool
}

File: assets/scripts/definitions/enemy.lua
---@meta

---@class enemy_intend_ctx
---@field type_id type_id
---@field guid guid
---@field round number

---Enemy represents a definition of a enemy that can be linked from a Actor.
---@class enemy
---@field id? type_id
---@field name string
---@field description string
---@field initial_hp number
---@field max_hp number
---@field look string
---@field color string
---@field intend? fun(ctx:enemy_intend_ctx):nil
---@field callbacks callbacks
---@field test? fun():nil|string
---@field base_game? boolean